File: src/breslin/clientside/game/states/GamePlay.java
package breslin.clientside.game.states;

/***************************************
*   		INCLUDES
***************************************/

//state machine
import breslin.statemachine.State;

//game
import breslin.clientside.game.Game;

public class GamePlay extends State
{

private Game mGame;

public GamePlay(Game game)
{
	mGame = game;
}

public void enter()
{

}

public void execute()
{
	//user input
	mGame.processInput();

	//network outgoing
	mGame.sendByteBuffer();

	//network ingoing
	mGame.checkForByteBuffer();
}

public void exit()
{

}

}

File: src/breslin/clientside/game/states/game_play.php
var GamePlay = new Class(
{

initialize: function(game)
{
	this.mGame = game; 
},

log: function(msg)
{
        setTimeout(function()
        {
                throw new Error(msg);
        }, 0);
},

enter: function()
{

},

execute: function()
{
	//user input
        this.mGame.processInput();

        //network outgoing
        this.mGame.sendByteBuffer();
	

	//don't need to do this as it's being done in main.html
       	/*	
	if (this.mGame.mPollDelayCounter > this.mGame.mPollDelay)
	{ 
		//network ingoing
        	this.mGame.checkForByteBuffer();
		
		//reset counter
		this.mGame.mPollDelayCounter = 0;
	}
	
	this.mGame.mPollDelayCounter++;	
	*/

},

exit: function()
{

}

});

